src/components/TransactionBlock/KlimaStakingTransactionBlock.tsx

```diff
@@ -279,10 +280,39 @@ const KlimaStakingTransactionBlock = ({
           return;
         }
 
+        const sdkChain = getSdkForChainId(CHAIN_ID.POLYGON);
+        const gasInfo = await sdkChain?.getGatewayGasInfo();
+
+        let priceUsd = await getAssetPriceInUsd(CHAIN_ID.POLYGON,ethers.constants.AddressZero,sdk);
```

Comment:
If either of these `await`'s return an error, what is the value of the error? As anything other than `null` would cause the next `if` statement to still pass...